torch/csrc/api/include/torch/nn/functional/embedding.h

```diff
@@ -1,12 +1,100 @@
 #pragma once
 
+#include <torch/nn/options/embedding.h>
+
 namespace torch {
 namespace nn {
 namespace functional {
 
 inline Tensor one_hot(const Tensor& tensor, int64_t num_classes = -1) {
   return torch::one_hot(tensor, num_classes);
 }
+
+inline Tensor embedding(const Tensor& input, Tensor weight, EmbeddingOptions options = {}) {
+  if (options.padding_idx() != c10::nullopt) {
+    if (*options.padding_idx() > 0) {
+      TORCH_CHECK(*options.padding_idx() < weight.size(0), "Padding_idx must be within num_embeddings");
+    }
+    else if (*options.padding_idx() < 0) {
+      TORCH_CHECK(*options.padding_idx() >= -weight.size(0), "Padding_idx must be within num_embedding");
+      options.padding_idx(weight.size(0) + *options.padding_idx());
+    }
+  } else {
+    options.padding_idx(-1);
+  }
+
+  if (options.max_norm() != c10::nullopt) {
```

Comment:
In Python there is `input = input.contiguous()` below this line, and I think we can pass `input` by value into this `embedding` function (i.e. making it `Tensor` instead of `const Tensor&`), so that we can call `input = input.contiguous()` here